CMakeLists.txt

```diff
@@ -63,6 +64,11 @@ if(CMAKE_Fortran_COMPILER_ID MATCHES "GNU")
 
   # Copied from MOM5/bin/mkmf.template.nci.gfortran
   set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -fcray-pointer -fdefault-real-8 -ffree-line-length-none -fno-range-check -Waliasing -Wampersand -Warray-bounds -Wcharacter-truncation -Wconversion -Wline-truncation -Wintrinsics-std -Wsurprising -Wno-tabs -Wunderflow -Wunused-parameter -Wintrinsic-shadow -Wno-align-commons")
+  # Note, -fallow-argument-mismatch is handled further down
+  check_fortran_compiler_flag("-fallow-invalid-boz" _boz_flag)
+  if(_boz_flag)
+    set(CMAKE_Fortran_FLAGS               "${CMAKE_Fortran_FLAGS} -fallow-invalid-boz" )
```

Comment:
Was  `fallow-invalid-boz` used in the MOM5 mkmf build? Why is it introduced here?

Reply:
Yes: https://github.com/ACCESS-NRI/spack-packages/blob/2ff2c584f60f872e61a6dfa5a3fe65a46f69b44b/packages/mom5/package.py#L161

Reply:
Thanks! Looks like I added `-fallow-argument-mismatch -fallow-invalid-boz` to the SPR, presumably to avoid build failures with newer versions of gfortran. However, it was never added to MOM5. Will you add it to your MOM5 CMake build and `bin/mkmf.template.ubuntu`?

Reply:
> Will you add it to your MOM5 CMake build 

Already there, see [here](https://github.com/ACCESS-NRI/MOM5/blob/51f67bf4a554d62f7b1ce99caf2f39dc1ef6b7b3/cmake/CMakeLists.txt#L63-L70).

> and `bin/mkmf.template.ubuntu` & `bin/mkmf.template.nci.gfortran`?

The comments in the SPR say the gcc flags were copied from `bin/mkmf.template.ubuntu` and the intel flags were copied from `bin/mkmf.template.nci`. I didn't realize you'd modified them and, now that I look, the comments seem to be incorrect anyway. Where exactly were things copied from?

Reply:
Great! The comments indicate where they original came from. The flags would have been modified as needed, including removing sections that were not required. Use a graphical diff program to see the differences more clearly.

Reply:
So you'd like me to add `-fallow-argument-mismatch -fallow-invalid-boz` to `bin/mkmf.template.ubuntu` on the master branch?

Reply:
Not my call, it's up to the Ocean Team to decide. You'd probably want to add it to any mkmf template that uses gcc. e.g. `bin/mkmf.template.ubuntu` and `bin/mkmf.template.nci.gfortran`.

Reply:
Added [here](https://github.com/ACCESS-NRI/MOM5/pull/39/commits/a4b3ae3f4d96120886e27a108b154e79346eae81)